src/mesh/mesh.cc

```diff
@@ -1074,17 +1064,11 @@ void Mesh::add_element(unsigned int elm_id, unsigned int dim, unsigned int regio
 	}
 	region_db_.mark_used_region(region_idx.idx());
 
-	if (region_idx.is_boundary()) {
-		bc_element_tmp_.push_back( ElementTmpData(elm_id, dim, region_idx, partition_id, node_ids) );
-	} else {
-		if(dim == 0 ) {
-			WarningOut().fmt("Bulk elements of zero size(dim=0) are not supported. Element ID: {}.\n", elm_id);
-		}
-		else {
-			Element *ele = add_element_to_vector(elm_id);
-			bulk_size_++;
-			this->init_element(ele, elm_id, dim, region_idx, partition_id, node_ids);
-		}
+	if (!region_idx.is_boundary() && dim == 0) {
```

Comment:
Do we still need separate `add_element_to_vector` and `add_element`? 
The reason was that `add_element` has created a temporary vector of boundary elements.

Reply:
`add_element` is a public function used by mesh readers. Mesh class uses the protected method `add_element_to_vector`.